frontend/src/lib/WorkflowParser.ts

```diff
@@ -168,49 +169,69 @@ export default class WorkflowParser {
   // Makes sure the workflow object contains the node and returns its
   // inputs/outputs if any, while looking out for any missing link in the chain to
   // the node's inputs/outputs.
-  public static getNodeInputOutputParams(workflow?: Workflow, nodeId?: string): [string[][], string[][]] {
-    type paramList = string[][];
+  public static getNodeInputOutputParams(workflow?: Workflow, nodeId?: string): [Array<KeyValue<string>>, Array<KeyValue<string>>] {
+    type paramList = Array<KeyValue<string>>;
     if (!nodeId || !workflow || !workflow.status || !workflow.status.nodes || !workflow.status.nodes[nodeId]) {
       return [[], []];
     }
 
     const node = workflow.status.nodes[nodeId];
     const inputsOutputs: [paramList, paramList] = [[], []];
     if (node.inputs && node.inputs.parameters) {
-      inputsOutputs[0] = node.inputs.parameters.map(p => [p.name, p.value || '']);
+      inputsOutputs[0] = node.inputs.parameters.map(p => [p.name, p.value || ''] );
     }
     if (node.outputs && node.outputs.parameters) {
-      inputsOutputs[1] = node.outputs.parameters.map(p => [p.name, p.value || '']);
+      inputsOutputs[1] = node.outputs.parameters.map(p => [p.name, p.value || ''] );
     }
     return inputsOutputs;
   }
 
+  // Makes sure the workflow object contains the node and returns its
+  // inputs/outputs artifacts if any, while looking out for any missing link in the chain to
+  // the node's inputs/outputs.
+  public static getNodeInputOutputArtifacts(workflow?: Workflow, nodeId?: string): [Array<KeyValue<S3Artifact>>, Array<KeyValue<S3Artifact>>] {
```

Comment:
nit: an object return value like `{ inputArtifacts, outputArtifacts }` clarifies both array contents better.

I can see you are following the same convention as `getNodeInputOutputParams` which is not a good example by itself.

Reply:
Also, please add a test similar to this one: https://github.com/kubeflow/pipelines/blob/cd0aeb6c6297de8dd58a979cd172cfb611bfa730/frontend/src/lib/WorkflowParser.test.ts#L339

Reply:
do u mean to change the return value from a tuple to an object?
or do u mean a tuple `[inputArtifacts, outputArtifacts]` ? with a type alias?

cuz changing the return value means a lot of changes downstream.

Reply:
Yes, I think changing the return value to an object makes usage code more readable. This has only one usage, so I believe it's fine.

Reply:
ok, done.
I also made the changes for `getNodeInputOutputParams`.
Updated the tests for both of them too.